components/promise-component.js

```diff
@@ -0,0 +1,66 @@
+// promise-component.js
+
+var _ = require('underscore');
+var noflo = require('noflo');
+
+module.exports = function(def){
+    if (_.isEmpty(def.inPorts)) throw Error("Missing inPorts");
+    if (def.resolvePort && !_.isString(def.resolvePort.name))
+        throw Error("Missing resolvePort name");
+    if (def.rejectPort && !_.isString(def.rejectPort.name))
+        throw Error("Missing rejectPort name");
+    var resolvePort = def.resolvePort || _.defaults({
```

Comment:
I'm not sure I understand your question. The code checks that if the caller provided a custom resolve/reject port, they must also provide a name; otherwise this code provides default resolve/reject ports.